tests/test_transcription.py

```diff
@@ -76,3 +76,131 @@ async def test_transcribe_prerecorded_file():
+
+    # Checks that the channels key contains the expected keys
+    assert len(response['results']['channels']) == 1
+
+    # Checks if alternatives is present in the channels[0] object
+    assert list(response['results']['channels'][0].keys()) == ['alternatives']
+
+    # Checks if alternatives is a list of length 1
+    assert len(response['results']['channels'][0]['alternatives']) == 1
+    
+    # Checks if the alternatives[0] object contains the expected keys
+    assert set(response['results']['channels'][0]['alternatives'][0].keys()) == set(['transcript', 'confidence', 'words', 'paragraphs'])
+
+    # Checks if the transcript is a string
+    assert type(response['results']['channels'][0]['alternatives'][0]['transcript']) == str
+
+    # Checks if the confidence is a float
+    assert type(response['results']['channels'][0]['alternatives'][0]['confidence']) == float
+
+    # Checks if the words is a list
+    assert type(response['results']['channels'][0]['alternatives'][0]['words']) == list
+
+    # Checks if the paragraphs is a dict
+    assert type(response['results']['channels'][0]['alternatives'][0]['paragraphs']) == dict
+
+    # Checks if the transcript within the paragraphs object is a string
+    assert type(response['results']['channels'][0]['alternatives'][0]['paragraphs']['transcript']) == str
+
+    # Checks if the paragraphs within the paragraphs object is a list
+    assert type(response['results']['channels'][0]['alternatives'][0]['paragraphs']['paragraphs']) == list
+
+    # Checks if the paragraphs[0] object contains the expected keys
+    assert set(response['results']['channels'][0]['alternatives'][0]['paragraphs']['paragraphs'][0]) == set(['sentences', 'num_words', 'start', 'end'])
+
+def test_diarization():
+
+    """
+    Testing the diarization output of the Deepgram response.
+    
+    """
+
+    response = deepgram.transcription.sync_prerecorded(
+        {
+            "url": AUDIO_URL
+        },
+        {
+            "model": "nova",
+            "smart_format": True,
+            "diarize": True
+        },
+    )
+
+    # Checks if the speakers key is present in the alternatives[0] object
+    assert set(response['results']['channels'][0]['alternatives'][0]['paragraphs']['paragraphs'][0].keys()) == set(['sentences','speaker', 'num_words', 'start', 'end'])
+
+def test_summarize():
+    """
+    Checking the summarize output of the Deepgram response, especially for summarize v2.
+    """
+    response = deepgram.transcription.sync_prerecorded(
+        {
+            "url": AUDIO_URL
+        },
+        {
+            "model": "nova",
+            "smart_format": True,
+            "summarize": 'v2'
+        },
+    )
+    
+    # Checks if the summary object has the expected keys
+    assert set(response['results']['summary'].keys()) == set(['result', 'short'])
+    # Check if the result is a string
+    assert type(response['results']['summary']['result']) == str
```

Comment:
Is it worth it to check if the value of this is `success` rather than only if it's a string? (I think it's `success` when the request succeeds.)

Reply:
Yup, ur right. Adding a case for this!